### ISSUE TITLE
setScriptTimeout() Fails: selenium-webdriver v3.4.0, Geckodriver v1.7.1, Firefox 56

### ISSUE THREAD

**Issue by gauntface** [mozilla/geckodriver#800]
In order to help us efficiently investigate your issue, please provide the following information:

## Platform and application details
* Platform: Linux - Ubuntu
* Firefox: 56 (54 and 55 work as expected)
* Selenium: NPM release 3.4.0

## Steps to reproduce
Create driver instance and run:

```javascript
driverRef.manage().timeouts().setScriptTimeout(2000);
```

- [ ] Reproducable testcase:

- [ ] A trace level log:

Node Trace:
```
WebDriverError: timeouts
      at Object.throwDecodedError (node_modules/selenium-webdriver/lib/error.js:489:11)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:519:13)
      at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:169:7)
  From: Task: WebDriver.manage().setTimeouts({script: 2000})
      at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:816:17)
      at legacyTimeout (node_modules/selenium-webdriver/lib/webdriver.js:1446:17)
      at driver_.schedule.catch (node_modules/selenium-webdriver/lib/webdriver.js:1400:25)
      at ManagedPromise.invokeCallback_ (node_modules/selenium-webdriver/lib/promise.js:1384:14)
      at TaskQueue.execute_ (node_modules/selenium-webdriver/lib/promise.js:3092:14)
      at TaskQueue.executeNext_ (node_modules/selenium-webdriver/lib/promise.js:3075:27)
      at asyncRun (node_modules/selenium-webdriver/lib/promise.js:2935:27)
      at node_modules/selenium-webdriver/lib/promise.js:676:7
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:169:7)
```

**Comment by whimboo** [comment]
Please attach a geckodriver trace log as requested in the new issue template.

**Comment by p0deje** [comment]
I see the same failure on Selenium CI: https://travis-ci.org/SeleniumHQ/selenium/jobs/251095621.

It can be easily reproduced with the following Ruby script:

```ruby
require 'selenium-webdriver'

Selenium::WebDriver.logger.level = :debug
options = Selenium::WebDriver::Firefox::Options.new
options.binary = '/Applications/FirefoxNightly.app/Contents/MacOS/firefox-bin'
options.log_level = 'trace'
driver = Selenium::WebDriver.for(:firefox, options: options)
driver.manage.timeouts.implicit_wait = 6
```

Here is the relevant part of log the comes from the last line of script:

```
2017-07-08 09:25:59 INFO Selenium -> POST session/6b192138-7b04-cd47-9903-3dd6c8a4a94b/timeouts
2017-07-08 09:25:59 INFO Selenium    >>> http://127.0.0.1:4444/session/6b192138-7b04-cd47-9903-3dd6c8a4a94b/timeouts | {"implicit":6000}
2017-07-08 09:25:59 DEBUG Selenium      > {"Accept"=>"application/json", "Content-Type"=>"application/json; charset=utf-8", "Content-Length"=>"17"}
1499484359092	webdriver::server	DEBUG	-> POST /session/6b192138-7b04-cd47-9903-3dd6c8a4a94b/timeouts {"implicit":6000}
1499484359100	geckodriver::marionette	TRACE	-> 34:[0,2,"timeouts",{"implicit":6000}]
1499484359102	Marionette	TRACE	0 -> [0,2,"timeouts",{"implicit":6000}]
1499484359104	Marionette	TRACE	0 <- [1,2,{"error":"unknown command","message":"timeouts","stacktrace":"WebDriverError@chrome://marionette/content/error.js:226:5\nUnknownCommandError@chrome://marionette/content/error.js:493:5\nexecute/req<@chrome://marionette/content/server.js:527:15\nTaskImpl_run@resource://gre/modules/Task.jsm:331:42\nTaskImpl@resource://gre/modules/Task.jsm:280:3\nasyncFunction@resource://gre/modules/Task.jsm:252:14\nTask_spawn@resource://gre/modules/Task.jsm:166:12\nexecute@chrome://marionette/content/server.js:524:15\nonPacket@chrome://marionette/content/server.js:495:7\n_onJSONObjectReady/<@chrome://marionette/content/transport.js:499:9\n"},null]
1499484359105	geckodriver::marionette	TRACE	<- [1,2,{"error":"unknown command","message":"timeouts","stacktrace":"WebDriverError@chrome://marionette/content/error.js:226:5\nUnknownCommandError@chrome://marionette/content/error.js:493:5\nexecute/req<@chrome://marionette/content/server.js:527:15\nTaskImpl_run@resource://gre/modules/Task.jsm:331:42\nTaskImpl@resource://gre/modules/Task.jsm:280:3\nasyncFunction@resource://gre/modules/Task.jsm:252:14\nTask_spawn@resource://gre/modules/Task.jsm:166:12\nexecute@chrome://marionette/content/server.js:524:15\nonPacket@chrome://marionette/content/server.js:495:7\n_onJSONObjectReady/<@chrome://marionette/content/transport.js:499:9\n"},null]
1499484359155	webdriver::server	DEBUG	<- 404 Not Found {"value":{"error":"unknown error","message":"timeouts","stacktrace":"stack backtrace:\n   0:        0x10f644444 - backtrace::backtrace::trace::h9f8fad52a1a8acf1\n   1:        0x10f6447fe - backtrace::capture::Backtrace::new::hbc1a12654c8fdba8\n   2:        0x10f56ef4d - webdriver::error::WebDriverError::new::h10bc8992bd1e5ce5\n   3:        0x10f57adc7 - geckodriver::marionette::MarionetteSession::response::h5d67b2b949a2fa47\n   4:        0x10f58876b - geckodriver::marionette::MarionetteConnection::send_command::h8c45fa72f63f91aa\n   5:        0x10f5799d8 - _$LT$geckodriver..marionette..MarionetteHandler$u20$as$u20$webdriver..server..WebDriverHandler$LT$geckodriver..marionette..GeckoExtensionRoute$GT$$GT$::handle_command::h46d5c164eeb39b93\n   6:        0x10f519e68 - _$LT$webdriver..server..Dispatcher$LT$T$C$$u20$U$GT$$GT$::run::ha1d1f891f84bb26d\n   7:        0x10f56fcea - webdriver::server::start::_$u7b$$u7b$closure$u7d$$u7d$::hfcc5d77f42486673\n   8:        0x10f563fea - _$LT$std..panic..AssertUnwindSafe$LT$F$GT$$u20$as$u20$core..ops..FnOnce$LT$$LP$$RP$$GT$$GT$::call_once::h92ea8c401572a582\n   9:        0x10f4a8dc9 - std::panicking::try::do_call::h9dc9800c79cf1d09\n  10:        0x10fad654a - __rust_maybe_catch_panic\n  11:        0x10f4a879c - std::panicking::try::h81921656202b0237\n  12:        0x10f4a5cb5 - std::panic::catch_unwind::h614633da02554547\n  13:        0x10f4a7d4c - std::thread::Builder::spawn::_$u7b$$u7b$closure$u7d$$u7d$::h856179287ee96795\n  14:        0x10f502ae6 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::h207273a6229b7fea\n  15:        0x10fad2714 - std::sys::imp::thread::Thread::new::thread_start::hae18fa578e305215\n  16:     0x7fffadb0993a - _pthread_body\n  17:     0x7fffadb09886 - _pthread_start"}}
2017-07-08 09:25:59 INFO Selenium <- {"value":{"error":"unknown error","message":"timeouts","stacktrace":"stack backtrace:\n   0:        0x10f644444 - backtrace::backtrace::trace::h9f8fad52a1a8acf1\n   1:        0x10f6447fe - backtrace::capture::Backtrace::new::hbc1a12654c8fdba8\n   2:        0x10f56ef4d - webdriver::error::WebDriverError::new::h10bc8992bd1e5ce5\n   3:        0x10f57adc7 - geckodriver::marionette::MarionetteSession::response::h5d67b2b949a2fa47\n   4:        0x10f58876b - geckodriver::marionette::MarionetteConnection::send_command::h8c45fa72f63f91aa\n   5:        0x10f5799d8 - _$LT$geckodriver..marionette..MarionetteHandler$u20$as$u20$webdriver..server..WebDriverHandler$LT$geckodriver..marionette..GeckoExtensionRoute$GT$$GT$::handle_command::h46d5c164eeb39b93\n   6:        0x10f519e68 - _$LT$webdriver..server..Dispatcher$LT$T$C$$u20$U$GT$$GT$::run::ha1d1f891f84bb26d\n   7:        0x10f56fcea - webdriver::server::start::_$u7b$$u7b$closure$u7d$$u7d$::hfcc5d77f42486673\n   8:        0x10f563fea - _$LT$std..panic..AssertUnwindSafe$LT$F$GT$$u20$as$u20$core..ops..FnOnce$LT$$LP$$RP$$GT$$GT$::call_once::h92ea8c401572a582\n   9:        0x10f4a8dc9 - std::panicking::try::do_call::h9dc9800c79cf1d09\n  10:        0x10fad654a - __rust_maybe_catch_panic\n  11:        0x10f4a879c - std::panicking::try::h81921656202b0237\n  12:        0x10f4a5cb5 - std::panic::catch_unwind::h614633da02554547\n  13:        0x10f4a7d4c - std::thread::Builder::spawn::_$u7b$$u7b$closure$u7d$$u7d$::h856179287ee96795\n  14:        0x10f502ae6 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::h207273a6229b7fea\n  15:        0x10fad2714 - std::sys::imp::thread::Thread::new::thread_start::hae18fa578e305215\n  16:     0x7fffadb0993a - _pthread_body\n  17:     0x7fffadb09886 - _pthread_start"}}
Selenium::WebDriver::Error::UnknownError: timeouts
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/response.rb:69:in `assert_ok'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/response.rb:32:in `initialize'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/http/common.rb:83:in `new'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/http/common.rb:83:in `create_response'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/http/default.rb:107:in `request'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/http/common.rb:61:in `call'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/bridge.rb:170:in `execute'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/w3c/bridge.rb:535:in `execute'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/w3c/bridge.rb:72:in `timeout'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/w3c/bridge.rb:63:in `implicit_wait_timeout='
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/common/timeouts.rb:32:in `implicit_wait='
	from (irb):7
	from /Users/p0deje/.rbenv/versions/2.4.1/bin/irb:11:in `<top (required)>'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/lib/bundler/cli/exec.rb:74:in `load'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/lib/bundler/cli/exec.rb:74:in `kernel_load'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/lib/bundler/cli/exec.rb:27:in `run'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/lib/bundler/cli.rb:360:in `exec'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/lib/bundler/vendor/thor/lib/thor.rb:369:in `dispatch'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/lib/bundler/cli.rb:20:in `dispatch'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/lib/bundler/vendor/thor/lib/thor/base.rb:444:in `start'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/lib/bundler/cli.rb:10:in `start'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/exe/bundle:35:in `block in <top (required)>'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/lib/bundler/friendly_errors.rb:121:in `with_friendly_errors'
	from /Users/p0deje/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.15.1/exe/bundle:27:in `<top (required)>'
	from /Users/p0deje/.rbenv/versions/2.4.1/bin/bundle:22:in `load'
	from /Users/p0deje/.rbenv/versions/2.4.1/bin/bundle:22:in `<main>'
```

**Comment by andreastt** [comment]
The `/session/{session id}/timeouts` endpoint was removed in https://bugzilla.mozilla.org/show_bug.cgi?id=1375425. The patch came with an accompanying fix to geckodriver, but a new version has not yet been released.

**Comment by andreastt** [comment]
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1379482 to track the 0.18.0 release.

**Comment by p0deje** [comment]
Looking forward for 0.18.0. Thank you!

**Comment by andreastt** [comment]
This should be fixed by the 0.18.0 release, which is now out.
